infra/cifuzz/cifuzz.py

```diff
@@ -124,35 +171,48 @@ def build_fuzzers(  # pylint: disable=too-many-arguments,too-many-locals
   out_dir = os.path.join(workspace, 'out')
   os.makedirs(out_dir, exist_ok=True)
 
-  # Build Fuzzers using docker run.
-  inferred_url, project_builder_repo_path = (
-      build_specified_commit.detect_main_repo(project_name,
-                                              repo_name=project_repo_name))
-  if not inferred_url or not project_builder_repo_path:
-    logging.error('Could not detect repo from project %s.', project_name)
-    return False
-  project_repo_name = os.path.basename(project_builder_repo_path)
-  src_in_project_builder = os.path.dirname(project_builder_repo_path)
-
-  manual_src_path = os.getenv('MANUAL_SRC_PATH')
-  if manual_src_path:
-    if not os.path.exists(manual_src_path):
+  project_src_path = get_project_src_path()
```

Comment:
nit: it's better practice to read all the env vars in one place (i.e. the main/entrypoints), then pass them as arguments to these functions.

It's easier to track where things come from, decouples environment requirements and make things easier to test.

Reply:
helper function for this would be nice

get_workspace_for_project

Reply:
Helper function: https://github.com/google/oss-fuzz/pull/4656/files#diff-789e2b62017a626b51b6e04c45e2175d141b2fc3a2beb942b47b495b185a948fR37

I now get all env vars mostly in one place, all in this file: https://github.com/google/oss-fuzz/pull/4656/files#diff-789e2b62017a626b51b6e04c45e2175d141b2fc3a2beb942b47b495b185a948fR30 which then calls into cifuzz.py